internal/pkg/deploy/cloudformation/stack/lb_web_svc.go

```diff
@@ -43,50 +44,53 @@ type LoadBalancedWebService struct {
 	manifest     *manifest.LoadBalancedWebService
 	httpsEnabled bool
 
+	// Fields for LoadBalancedWebService that needs a Network Load Balancer.
+	dnsDelegationEnabled   bool
+	publicSubnetCIDRBlocks []string
+
 	parser loadBalancedWebSvcReadParser
 }
 
-// NewLoadBalancedWebService creates a new LoadBalancedWebService stack from a manifest file.
-func NewLoadBalancedWebService(mft *manifest.LoadBalancedWebService, env, app string, rc RuntimeConfig) (*LoadBalancedWebService, error) {
-	parser := template.New()
-	addons, err := addon.New(aws.StringValue(mft.Name))
-	if err != nil {
-		return nil, fmt.Errorf("new addons: %w", err)
-	}
-	return &LoadBalancedWebService{
-		ecsWkld: &ecsWkld{
-			wkld: &wkld{
-				name:   aws.StringValue(mft.Name),
-				env:    env,
-				app:    app,
-				rc:     rc,
-				image:  mft.ImageConfig.Image,
-				parser: parser,
-				addons: addons,
-			},
-			logRetention:        mft.Logging.Retention,
-			tc:                  mft.TaskConfig,
-			taskDefOverrideFunc: override.CloudFormationTemplate,
-		},
-		manifest:     mft,
-		httpsEnabled: false,
-
-		parser: parser,
-	}, nil
+// NewHTTPLoadBalancedWebService creates a new LoadBalancedWebService stack from a manifest file.
+func NewHTTPLoadBalancedWebService(mft *manifest.LoadBalancedWebService, env, app string, rc RuntimeConfig) (*LoadBalancedWebService, error) {
+	return newLoadBalancedWebService(mft, env, app, rc)
 }
 
 // NewHTTPSLoadBalancedWebService  creates a new LoadBalancedWebService stack from its manifest that needs to be deployed to
 // a environment within an application. It creates an HTTPS listener and assumes that the environment
 // it's being deployed into has an HTTPS configured listener.
 func NewHTTPSLoadBalancedWebService(mft *manifest.LoadBalancedWebService, env, app string, rc RuntimeConfig) (*LoadBalancedWebService, error) {
-	webSvc, err := NewLoadBalancedWebService(mft, env, app, rc)
+	webSvc, err := NewHTTPLoadBalancedWebService(mft, env, app, rc)
 	if err != nil {
 		return nil, err
 	}
 	webSvc.httpsEnabled = true
 	return webSvc, nil
 }
 
+// NewNetworkLoadBalancedWebService creates a new LoadBalancedWebService stack from a manifest file. It creates a
+// Network Load Balancer, with or without an Application Load Balancer.
```

Comment:
What do you think of this alternative API:
1. We have a single constructor `NewLoadBalancedWebService`
2. We have many [functional options](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis)?
```go
func NewLoadBalancedWebService(mft *manifest.LoadBalancedWebService, env, app string, rc RuntimeConfig, opts  ...LoadBalancedWebServiceOption) (*LoadBalancedWebService, error)

type LoadBalancedWebServiceOption func(lbws *LoadBalancedWebService)

func WithNLB(cidrBlocks) LoadBalancedWebServiceOption { ... }
func WithNLBAndTLS(cidrBlocks) LoadBalancedWebServiceOption { ... }
```